tests/testthat/test-bind.R

```diff
@@ -517,3 +517,27 @@ test_that("rbind repairs names of data frames (#704)", {
     class = "vctrs_error_names_must_be_unique"
   )
 })
+
+test_that("rbind supports names and inner names (#689)", {
```

Comment:
This _should_ work with S3 objects like factors / POSIXct that `vec_set_names()` has to fall back to `names<-` on, but could you please add a test for that too? It would make me feel a bit better about it

Reply:
This would have caught the unassigned return you mentioned above, so that test is definitely needed!